### DISCUSSION TITLE
Global events to log exceptions

### DISCUSSION THREAD

**Discussion by gulshan** [dotnet/csharplang#7518] [Language Ideas]
The most common thing to do with the Exceptions is to log them. It can be called one of those "cross-cutting concerns". How about a few global events to do just that? There can be events for "All", "Caught" and "Uncaught" exceptions. These events can be fired on the catch-site or crash-site of the exceptions.

To be clear, I think the events should just have a read-only access of the exceptions. Meaning the events should not be able to "handle" or "consume" them, only log/record them. So, it would not prevent a crash or affect current exception handling mechanism in any way. But it will provide an additional centralized way to log the exceptions. Also, recording uncaught exceptions can be a fairly useful new feature.

I also think that these events should only be available to "executable" assemblies, not the libraries. Also, for the "caught" exceptions, the catch-site should be limited to the same assembly only, not the libraries.

Something similar can be found in the JS ecosystems.

**Comment by CyrusNajmabadi** [comment]
This is not a language issue.  Closing out.  Request for the runtime, and the behavior there can be at dotnet/runtime.  Thanks!

**Reply by gulshan** [reply]
Done https://github.com/dotnet/runtime/discussions/91752
